src/realm/object-store/c_api/app.cpp

```diff
@@ -253,6 +253,25 @@ RLM_API realm_app_t* realm_app_create(const realm_app_config_t* app_config,
     });
 }
 
+RLM_API realm_app_t* realm_app_create_cached(const realm_app_config_t* app_config,
+                                             const realm_sync_client_config_t* sync_client_config)
+{
+    return wrap_err([&] {
+        return new realm_app_t(App::get_shared_app(*app_config, *sync_client_config));
+    });
+}
+
+RLM_API bool realm_app_get_cached(const char* app_id, const char* base_url, realm_app_t** out_app)
+{
+    return wrap_err([&] {
+        auto app =
+            App::get_cached_app(std::string(app_id), base_url ? util::some<std::string>(base_url) : util::none);
+        set_out_param(out_app, app ? new realm_app_t(app) : nullptr);
```

Comment:
Can you change this to: 
```
if(out_app)
    *out_app = app ? new realm_app_t{app} : nullptr; 
```
Same as we do for `realm_app_switch_user`.
The reason for this is that, `set_out_param` is supposed to initialize a piece of memory that has been already allocated (It accepts T* and not T**), in this API you need to allocated and initialize the memory. 
The code as it is may or may not be successful.
Thanks.

Reply:
I have to admit pointers in C++ are not my strong suit and it's very possible I'm missing something, but fundamentally, it seems to me that this code should work - it's just that `T` in our case is `realm_app_t*`, meaning `T*` is `realm_app_t**`. The memory pointed to by the first pointer in `realm_app_t*` is already allocated by the caller, which is validated by the `if (out_n)` check in `set_out_param`.

To be clear, I don't mind making the change, I'm just trying to understand what am I missing.